drivers/ft5x06/ft5x06_touch_dev.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2021 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_ft5x06
+ * @{
+ *
+ * @file
+ * @brief       Driver adaption to touch_dev generic interface
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ * @}
+ */
+
+#include <stddef.h>
+#include <stdint.h>
+#include <errno.h>
+#include <assert.h>
+
+#include "kernel_defines.h"
+#include "periph/gpio.h"
+
+#include "ft5x06.h"
+#include "ft5x06_touch_dev.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+uint16_t _ft5x06_height(const touch_dev_t *touch_dev)
+{
+    const ft5x06_t *dev = (const ft5x06_t *)touch_dev;
+    assert(dev);
+
+    return dev->params.ymax;
+}
+
+uint16_t _ft5x06_width(const touch_dev_t *touch_dev)
+{
+    const ft5x06_t *dev = (const ft5x06_t *)touch_dev;
+    assert(dev);
+
+    return dev->params.xmax;
+}
+
+static ft5x06_touch_position_t positions[FT5X06_TOUCHES_COUNT_MAX];
+
+uint8_t _ft5x06_touches(const touch_dev_t *touch_dev, touch_t *touches, size_t len)
+{
+    ft5x06_t *dev = (ft5x06_t *)touch_dev;
+    assert(dev);
+
+    uint8_t count;
+    ft5x06_read_touch_count(dev, &count);
+    uint8_t ret = (count > 0);
+
+    /* Ensure we have enough space to store the detected touches */
+    assert(len >= count);
+
+    if (ret && touches != NULL) {
+        ft5x06_read_touch_positions(dev, positions, count);
+        for (uint8_t touch = 0; touch < count; touch++) {
+            touches[touch].x = positions[touch].x;
+            touches[touch].y = positions[touch].y;
```

Comment:
Why can't we read them directly into `touches`?

Reply:
>  Why can't we read them directly into touches?

We can indeed :)